src/apps/DesktopAppPage.vala

```diff
@@ -192,27 +321,29 @@ namespace Ilia {
          * that extracts the necessary data from the TreeModel and TreeIter objects
          */
         private int app_sort_func(TreeModel model, TreeIter a, TreeIter b) {
-            string query_string = entry.get_text().down();
-
             DesktopAppInfo app_a;
             model.@get(a, ITEM_VIEW_COLUMN_APPINFO, out app_a);
             DesktopAppInfo app_b;
             model.@get(b, ITEM_VIEW_COLUMN_APPINFO, out app_b);
 
+            // Check if either app is a favorite
+            bool is_favorite_a;
+            model.@get(a, ITEM_VIEW_COLUMN_IS_FAVORITE, out is_favorite_a);
+            bool is_favorite_b;
+            model.@get(b, ITEM_VIEW_COLUMN_IS_FAVORITE, out is_favorite_b);
+
+            // Favorites go on top
+            if (is_favorite_a && !is_favorite_b) {
+                return -1;
+            } else if (!is_favorite_a && is_favorite_b) {
+                return 1;
+            }
+
+            // If both are favorites or both are not favorites, sort alphabetically first
             var app_a_name = app_a.get_name().down();
             var app_b_name = app_b.get_name().down();
-
-            var app_a_id = app_a.get_id();
-            var app_b_id = app_b.get_id();
-
-            return compare_desktop_apps(
-                app_a_name,
-                app_b_name,
-                app_a_id,
-                app_b_id,
-                query_string,
-                launch_counts
-            );
```

Comment:
I guess we could still have used this. favorites should be at the top, then apps sorted by launch counts and then alphabetical sort.